### PR TITLE
Build TensorFlow with x10 support if building an x10 toolchain.

### PR DESCRIPTION
**Pull request by pschuh**

### CODE REVIEW

File: Sources/TensorFlow/CMakeLists.txt [hunk 1]
```diff
@@ -63,6 +62,22 @@ add_library(TensorFlow SHARED
...
+  ../x10/swift_bindings/apis/RawOpsManual.swift
+
+  ../x10/swift_bindings/Device.swift
+  ../x10/swift_bindings/XLAScalarType.swift
+  ../x10/swift_bindings/XLATensor.swift)
+target_compile_definitions(TensorFlow PRIVATE
+  USING_X10_BACKEND DEFAULT_BACKEND_EAGER)
+else()
+target_sources(TensorFlow PRIVATE
+  Core/Device.swift)
```
Comment (compnerd) [inline Sources/TensorFlow/CMakeLists.txt:79]: Can you indent this a level please?

File: Sources/TensorFlow/CMakeLists.txt [hunk 2]
```diff
@@ -63,6 +62,22 @@ add_library(TensorFlow SHARED
...
+target_sources(TensorFlow PRIVATE
+  ../x10/swift_bindings/apis/CrossReplicaSum.swift
+  ../x10/swift_bindings/apis/DeviceScope.swift
+  ../x10/swift_bindings/apis/RawOpsManual.swift
+
+  ../x10/swift_bindings/Device.swift
+  ../x10/swift_bindings/XLAScalarType.swift
+  ../x10/swift_bindings/XLATensor.swift)
+target_compile_definitions(TensorFlow PRIVATE
+  USING_X10_BACKEND DEFAULT_BACKEND_EAGER)
```
Comment (compnerd) [inline Sources/TensorFlow/CMakeLists.txt:76]: Can you indent the body of the `if` a level please?

**Comment by asuhan** [comment]
Test failures look relevant.

File: Sources/TensorFlow/CMakeLists.txt [hunk 3]
```diff
@@ -63,6 +62,21 @@ add_library(TensorFlow SHARED
...
+  target_sources(TensorFlow PRIVATE
+    ../x10/swift_bindings/apis/CrossReplicaSum.swift
+    ../x10/swift_bindings/apis/DeviceScope.swift
+    ../x10/swift_bindings/apis/RawOpsManual.swift
+
+    ../x10/swift_bindings/Device.swift
+    ../x10/swift_bindings/XLAScalarType.swift
+    ../x10/swift_bindings/XLATensor.swift)
+  target_compile_definitions(TensorFlow PRIVATE
+    USING_X10_BACKEND DEFAULT_BACKEND_EAGER)
```
Comment (compnerd) [inline Sources/TensorFlow/CMakeLists.txt:75]: Nit: might be nice to add a newline instead of space.

**Pull request merged**